src/Avalonia.Themes.Fluent/Controls/SplitButton.xaml

```diff
@@ -0,0 +1,205 @@
+    <Setter Property="BorderBrush" Value="{DynamicResource SplitButtonBorderBrushChecked}" />
+    <Setter Property="TextBlock.Foreground" Value="{DynamicResource SplitButtonForegroundChecked}" />
+  </Style>
+  <Style Selector="SplitButton:checked /template/ Button#PART_SecondaryButton PathIcon">
+    <Setter Property="Foreground" Value="{DynamicResource SplitButtonForegroundChecked}" />
+  </Style>
+
+  <!-- Checked PointerOver State -->
+  <Style Selector="SplitButton:checked /template/ Button#PART_PrimaryButton:pointerover /template/ ContentPresenter,
+         SplitButton:checked /template/ Button#PART_SecondaryButton:pointerover /template/ ContentPresenter">
+    <Setter Property="Background" Value="{DynamicResource SplitButtonBackgroundCheckedPointerOver}" />
+    <Setter Property="BorderBrush" Value="{DynamicResource SplitButtonBorderBrushCheckedPointerOver}" />
+    <Setter Property="TextBlock.Foreground" Value="{DynamicResource SplitButtonForegroundCheckedPointerOver}" />
+  </Style>
+  <Style Selector="SplitButton:checked /template/ Button#PART_SecondaryButton:pointerover PathIcon">
+    <Setter Property="Foreground" Value="{DynamicResource SplitButtonForegroundCheckedPointerOver}" />
+  </Style>
+
+  <!-- Checked Pressed State -->
+  <Style Selector="SplitButton:checked /template/ Button#PART_PrimaryButton:pressed /template/ ContentPresenter,
+         SplitButton:checked /template/ Button#PART_SecondaryButton:pressed /template/ ContentPresenter">
+    <Setter Property="Background" Value="{DynamicResource SplitButtonBackgroundCheckedPressed}" />
+    <Setter Property="BorderBrush" Value="{DynamicResource SplitButtonBorderBrushCheckedPressed}" />
+    <Setter Property="TextBlock.Foreground" Value="{DynamicResource SplitButtonForegroundCheckedPressed}" />
+  </Style>
+  <Style Selector="SplitButton:checked /template/ Button#PART_SecondaryButton:pressed PathIcon">
+    <Setter Property="Foreground" Value="{DynamicResource SplitButtonForegroundCheckedPressed}" />
+  </Style>
+
+  <!-- Checked Flyout Open State -->
+  <Style Selector="SplitButton:checked:flyout-open /template/ Button#PART_PrimaryButton /template/ ContentPresenter,
+         SplitButton:checked:flyout-open /template/ Button#PART_SecondaryButton /template/ ContentPresenter">
+    <Setter Property="Background" Value="{DynamicResource SplitButtonBackgroundCheckedPressed}" />
+    <Setter Property="BorderBrush" Value="{DynamicResource SplitButtonBorderBrushCheckedPressed}" />
+    <Setter Property="TextBlock.Foreground" Value="{DynamicResource SplitButtonForegroundCheckedPressed}" />
+  </Style>
+  <Style Selector="SplitButton:checked:flyout-open /template/ Button#PART_SecondaryButton PathIcon">
+    <Setter Property="Foreground" Value="{DynamicResource SplitButtonForegroundCheckedPressed}" />
+  </Style>
+
+  <!-- Disabled Checked State -->
+  <Style Selector="SplitButton:checked:disabled /template/ Button#PART_PrimaryButton /template/ ContentPresenter,
```

Comment:
Chaining selectors toghether like this `SplitButton:checked:disabled` does have downsides as far as I can tell: The :checked selector will still be matched and those setters applied. Then the :checked:disabled will also be run and applied overwriting any changed done by the first. That is unnecessary obviously and negatively impacts performance to some small degree. In addition you are making the order of styles important which isn't always intuitive for those coming over from other XAML frameworks.

I'll address of all your style comments here as the same ideas apply.

The 'Avalonia' methadology has the following downsides:
 1. [Con] It is uselessly applying setters only to have the same properties overwritten later when a mor specific match is made. That seems like a fundamental downside to chaining together like this -- performance.
 2. [Con] As mentioned order of styles becomes very important. That is just something to get used to I suppose but I wouldn't say it is better than other XAML style systems -- it's worse in my opinion.
 3. [Con] You are actually making it more difficult to change/edit things in the future. It is more complex to keep track of things.
 4. [Pro] Reduces the number of pseudoclasses in code behind considerably. Marginally simplifies the styles as well.


With the original implementation:
 1. [Con] Lots of pseudoclasses which may not look correct for Avalonia's conventions
 2. [Pro] Easy to understand and modify (there are no gotchas or complex selectors)
 3. [Pro] Technically it is faster as only one 'state' is every applied. There is no risk of uselessly applying setters just for them to be overwritten later.
 4. [Pro] Styles are NOT order dependent
 5. [Pro] It was still based on upstream so any changes there could relatively easily flow back into Avalonia. That definitely counts for something considering how widely used WinUI/UWP is. 

 If it was me, you are simply moving complexity. 

Avalonias Styling system is not complete yet and I and others have made several issues about that. I can't say right now which is really better here. It seems open to debate.

Reply:
Bottom line: you are moving the complexity from the code-behind and pseudoclasses into the Style selectors. The reasoning for this is it's Avalonia convention.

Subjectively speaking the selectors are harder to understand and maintain. Objectively speaking I still expect there is a performance penalty, perhaps offset by combining selectors like you did though. With the above mentioned pros/cons lists clearly defined; overall, I guess I'm not a big fan of the way this works. Obviously, its your call though and it does match Avalonia convention which is what you will go with. 

As you said, when the styling system implements elsewhere mentioned improvements some concerns will improve.